pkg/helm/nelm/nelm.go

```diff
@@ -0,0 +1,254 @@
+	if err != nil {
+		// FIXME(ilya-lesikov): return a specific error from ReleaseGet if release not found
+		if regexp.MustCompile(`.+release ".+" (namespace ".+") not found`).Match([]byte(err.Error())) {
+			return "0", "", fmt.Errorf("get nelm release %q: %w", releaseName, err)
+		}
+
+		return "", "", fmt.Errorf("get nelm release %q: %w", releaseName, err)
+	}
+
+	return strconv.FormatInt(int64(releaseGetResult.Release.Revision), 10), releaseGetResult.Release.Status.String(), nil
+}
+
+func (c *NelmClient) UpgradeRelease(releaseName string, chartName string, valuesPaths []string, setValues []string, labels map[string]string, namespace string) error {
+	c.logger.Info("Running nelm install for release",
+		slog.String("release", releaseName),
+		slog.String("chart", chartName),
+		slog.String("namespace", namespace))
+
+	// XXX: for some reason upgrade was called twice previously?
+	// XXX: why we always deployed with upg.SkipCRDs = true?
+	// XXX: Nelm does not care about pending status and will upgrade pending release just fine. Does it mean we are going to be fine without this: https://github.com/flant/addon-operator/blob/b7c97010128922e066fed35e2e53e91834a250de/pkg/helm/helm3lib/helm3lib.go#L293-L300
+	// FIXME(ilya-lesikov): allow passing labels to Release.Labels
+	if err := action.ReleaseInstall(commonOptions.context, releaseName, namespace, action.ReleaseInstallOptions{
+		// XXX: Nelm doesn't allow anything besides chart dir
+		ChartDirPath:         chartName,
+		ExtraLabels:          c.labels,
+		KubeContext:          commonOptions.KubeContext,
+		ReleaseHistoryLimit:  int(commonOptions.HistoryMax),
+		ReleaseStorageDriver: commonOptions.HelmDriver,
+		ValuesFilesPaths:     valuesPaths,
+		ValuesSets:           setValues,
+	}); err != nil {
+		return fmt.Errorf("install nelm release %q: %w", releaseName, err)
+	}
+
+	c.logger.Info("Nelm install successful",
+		slog.String("release", releaseName),
+		slog.String("chart", chartName),
+		slog.String("namespace", namespace))
+
+	return nil
+}
+
+func (c *NelmClient) GetReleaseValues(releaseName string) (utils.Values, error) {
+	panic("not going to be implemented")
+}
+
+func (c *NelmClient) GetReleaseChecksum(releaseName string) (string, error) {
+	releaseGetResult, err := action.ReleaseGet(commonOptions.context, releaseName, commonOptions.Namespace, action.ReleaseGetOptions{
+		KubeContext:          commonOptions.KubeContext,
+		OutputNoPrint:        true,
+		ReleaseStorageDriver: commonOptions.HelmDriver,
+	})
+	if err != nil {
+		return "", fmt.Errorf("get nelm release %q: %w", releaseName, err)
+	}
+
+	if checksum, ok := releaseGetResult.Release.Annotations["moduleChecksum"]; ok {
+		return checksum, nil
+	}
+
+	// XXX: here was a fallback to get "_addonOperatorModuleChecksum" value from last release values
+	// instead. In Nelm via action.ReleaseGet we don't expose values of last release, I don't think
+	// they are relevant when we already expose manifests. Do we really need this?
```

Comment:
We can't ignore it. It's just a migration while we try to move from old values-injected method to release-metadata method. I guess we will finish this migration before the nelm engine will be enabled by default

Reply:
Going to expose Values and do the fallback then.